hta/common/trace.py

```diff
@@ -470,8 +472,17 @@ def parse_multiple_ranks(
             logger.debug(f"finished parsing for all {len(ranks)} ranks")
         else:
             num_procs = min(mp.cpu_count(), len(ranks))
+            if len(ranks) <= 4:
```

Comment:
What is the reasoning behind using `len(ranks) <= 4`? If I recall correctly, it was 8 earlier which I assume is due to the fact that node/server has 8 GPUs generally.